src/gui/render/graphics_shaders/program_handler.hpp

```diff
@@ -0,0 +1,436 @@
+// -*- lsst-c++ -*-
+/*
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ */
+
+/**
+ * @file program_handler.hpp
+ *
+ * @author @AlemSnyder
+ *
+ * @brief Define ShaderHandler
+ *
+ * @ingroup GUI  RENDER  GRAPHICS_SHADERS
+ *
+ */
+
+#pragma once
+
+#include "logging.hpp"
+#include "opengl_program_status.hpp"
+#include "types.hpp"
+
+#include <GL/glew.h>
+
+#include <compare>
+#include <filesystem>
+#include <map>
+#include <optional>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#define NO_UNIFORM_INT -1
+
+namespace gui {
+
+namespace shader {
+
+/**
+ * @brief File that can be opened.
+ *
+ * @details Has the file path, and stores the status of the file.
+ */
+class File {
+ private:
+    // path to file
+    const std::filesystem::path file_;
+    // file status
+    FileStatus status_;
+
+ public:
+    /**
+     * @brief Construct a new File object.
+     *
+     * @param file file path
+     */
+    inline File(const std::filesystem::path file) :
+        file_(std::filesystem::absolute(file)), status_(FileStatus::EMPTY) {}
+
+    /**
+     * @brief Returns file contents as a string.
+     *
+     * @return std::optional<std::string> file contents
+     */
+    [[nodiscard]] std::optional<std::string> get_file_content();
+
+    /**
+     * @brief Get file status.
+     *
+     * @return FileStatus
+     */
+    [[nodiscard]] inline FileStatus
+    get_status() const {
```

Comment:
```suggestion
    get_error() const {
```

Reply:
not done

Reply:
I don't want to rename it. It dosn't seem like a big deal to me.